api/src/com/cloud/network/vpc/OSPFZoneConfig.java

```diff
@@ -0,0 +1,329 @@
+    }
+
+    public Short getRetransmitInterval() {
+        return retransmitInterval;
+    }
+
+    public void setRetransmitInterval(Short retransmitInterval) {
+        this.retransmitInterval = retransmitInterval;
+    }
+
+    public Short getTransitDelay() {
+        return transitDelay;
+    }
+
+    public void setTransitDelay(Short transitDelay) {
+        this.transitDelay = transitDelay;
+    }
+
+    public Authentication getAuthentication() {
+        return authentication;
+    }
+
+    public void setAuthentication(Authentication authentication) {
+        this.authentication = authentication;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public CIDR[] getSuperCIDR() {
+        return superCIDRList;
+    }
+
+    public void setSuperCIDR(CIDR[] superCIDR) {
+        this.superCIDRList = superCIDR;
+    }
+
+    public Boolean getEnabled() {
+        return enabled;
+    }
+
+    public void setEnabled(Boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    public enum Params {
+        PROTOCOL, AREA, HELLO_INTERVAL, DEAD_INTERVAL, RETRANSMIT_INTERVAL, TRANSIT_DELAY, AUTHENTICATION, SUPER_CIDR, PASSWORD, ENABLED
+    }
+
+    public enum Protocol {
+        OSPF
+    }
+
+    public enum Authentication {
+        MD5, PLAIN_TEXT
+    }
+
+    public OSPFZoneConfig() {
+    }
+
+    public void setDefaultValues(final Map<String, String> details) throws BadCIDRException {
+        this.protocol = details.get(Params.PROTOCOL.name()) == null ? Protocol.OSPF : Protocol.valueOf(details.get(Params.PROTOCOL.name()));
+        this.ospfArea = details.get(Params.AREA.name()) == null ? "0" : details.get(Params.AREA.name());
+        this.helloInterval = Short.valueOf(details.get(Params.HELLO_INTERVAL.name()) == null ? "10" : details.get(Params.HELLO_INTERVAL.name()));
+        this.deadInterval = Short.valueOf(details.get(Params.DEAD_INTERVAL.name()) == null ? "40" : details.get(Params.DEAD_INTERVAL.name()));
+        this.retransmitInterval = Short.valueOf(details.get(Params.RETRANSMIT_INTERVAL.name()) == null ? "5" : details.get(Params.RETRANSMIT_INTERVAL.name()));
+        this.transitDelay = Short.valueOf(details.get(Params.TRANSIT_DELAY.name()) == null ? "1" : details.get(Params.TRANSIT_DELAY.name()));
+        this.authentication = details.get(Params.AUTHENTICATION.name()) == null ? Authentication.MD5 : Authentication.valueOf(details.get(Params.AUTHENTICATION.name()));
+        this.password = details.get(Params.PASSWORD.name()) == null ? "" : details.get(Params.PASSWORD.name());
+        this.superCIDRList = details.get(Params.SUPER_CIDR.name()) == null ? new CIDR[0] : NetUtils.convertToCIDR(details.get(Params.SUPER_CIDR.name()).split(","));
+        this.enabled = details.get(Params.ENABLED.name()) == null ? Boolean.FALSE : Boolean.valueOf(details.get(Params.ENABLED.name()));
```

Comment:
There is a repeating group of the following logic:

```details.get(<key>) == null ? <default value> : details.get(<key>)```

Please extract to utility method in CloudStack comment (e.g. ``<K, V> V getOptionalValue(final Map<K, V> map, K key, V default)``) with the appropriate unit tests.  Also, consider using ``Map.containsKey`` to check for key existence.